opentracing-spring-web/src/main/java/io/opentracing/contrib/spring/web/client/decorator/RestTemplateHeaderSpanDecorator.java

```diff
@@ -0,0 +1,100 @@
+package io.opentracing.contrib.spring.web.client.decorator;
+
+import static org.springframework.util.StringUtils.hasText;
+
+import io.opentracing.Span;
+import io.opentracing.contrib.spring.web.client.RestTemplateSpanDecorator;
+import io.opentracing.tag.StringTag;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.core.Ordered;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpResponse;
+
+/**
+ * RestTemplateHeaderSpanDecorator will decorate the span based on HTTP headers of the request.
+ * Prior to the request, the headers are compared to the {@link #allowedHeaders} list, if it's part
+ * of the provided list, they will be added as {@link StringTag}.
+ * The tag format will be a concatenation of {@link #prefix} and {@link HeaderEntry#tag}
+ */
+public class RestTemplateHeaderSpanDecorator  implements RestTemplateSpanDecorator, Ordered {
```

Comment:
It's probably not mandatory but it #85 allow people that are using the library to easily include there own SpanDecorator (see the usage of `ObjectProvider`) and it allows us to sort all of them in the autoconfigure like this: https://github.com/opentracing-contrib/java-spring-web/pull/85/files#diff-64700c454a15ffea7ba50cd59314a697R64

If we dont put any number, the default value is the following which makes it harder to run another decorator before/after this one:
>Any object that does not provide its own order value is implicitly assigned a value of Ordered.LOWEST_PRECEDENCE, thus ending up at the end of a sorted collection in arbitrary order with respect to other objects with the same order value.

Reply:
What is the use case for sorting span decorators?

Reply:
I'd say it could be useful if 2 decorators set the same tag, with that configuration you can decide which one will set it last.

From what I see we use it to make the MDC decorator run before the Header decorator.

It could be removed from the opensource version of it

Reply:
Sorry, it is still not sure why the order is necessary. 

The decorator operates on Span and request. OT spec does not say how should implementation handle multiple tags with the same key.

Reply:
I would rather remove it and add it later if needed. Could you please also rebase?

Reply:
I'm so sorry for this being not related to the PR, but I got really curious. You said that `It could be removed from the opensource version of it`, so I got this question: does opentracing has a paid version? If so, could you provide me some links and benefits for using a paid version? Thanks and again, sorry for putting this inside a PR, I was just "reviewing" the code and this really got my attention.

Reply:
There is no paid version. However, companies can have private forks with add functionality which is related yo their use cases and do not widely apply to everybody.